Книга Сергей

rybufc opened the pull request:
[Посмотреть решение](https://rawgit.com/rybufc/markup-task-2/master/tasks/1.html)

rybufc committed:
решение

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

rybufc committed:
style fixes

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

rybufc committed:
Пофиксил 11 задание с учётом уточнения из слака.

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

Shaikhislamov commented on tasks/1.html:
```diff
@@ -12,7 +13,25 @@ <h1 class="global-heading">&nbsp;</h1>
+                        background: #98ff66;
+                    }
+
+                    aside
```
у `div` и `aside` одинаковые стили, можно объединить в 1 правило

shaikma commented on tasks/1.html:
```diff
@@ -12,7 +13,25 @@ <h1 class="global-heading">&nbsp;</h1>
+                        background: #6698ff;
+                    }
+
+                    aside
```
у div и aside одинаковые стили, можно объединить в 1 правило

shaikma commented on tasks/11.html:
```diff
@@ -11,7 +12,20 @@ <h1 class="global-heading">&nbsp;</h1>
+                    border: 3px solid #98ff66;
+                }
+
+                article > input:hover
```
в задании написано что рамка должна быть при фокусе, а не ховере

shaikma commented on tasks/11.html:
```diff
@@ -11,7 +12,20 @@ <h1 class="global-heading">&nbsp;</h1>
         <main class="global-results">
             <style>
-                /* your styles here */
+                article > input:first-child
```
цвет рамки зависит от того, проходят ли валидацию введенные пользователям значения, посмотри какие псевдокласса бывают у форм и инпутов

shaikma commented on tasks/13.html:
```diff
@@ -11,7 +12,24 @@ <h1 class="global-heading">&nbsp;</h1>
+                ff6698
+                */
+
+                .global-scope > div:first-child > div
```
посмотри внимательнее на покрашенные теги и попробуй найти закономерность, возможно какой-то псевдоселектор подойдет)

shaikma commented on tasks/16.html:
```diff
@@ -11,7 +12,25 @@ <h1 class="global-heading">&nbsp;</h1>
-                /* your styles here */
+                article div:first-child
+                {
+                    background: #ff6698 !important;
```
кажется `!important` тут не нужен

shaikma commented on tasks/16.html:
```diff
@@ -11,7 +12,25 @@ <h1 class="global-heading">&nbsp;</h1>
+
+                .c1
+                {
+                    background: #6698ff !important;
```
от этого тоже можно избавиться

shaikma commented on tasks/17.html:
```diff
@@ -11,7 +12,15 @@ <h1 class="global-heading">&nbsp;</h1>
+
+                input:checked
+                {
+                    outline: none !important;
```
![cdd3e1f4e36a05351f37d38bcbb07e25b63ab39fa4351c4497a0aa2b26ae7fe3](https://user-images.githubusercontent.com/13517829/31681409-d21c4302-b37f-11e7-9e3e-15eee30be7f7.jpg)

shaikma replied:
использовать без веской на то причины `!important` не стоит , тут волне можно обойтись без него

shaikma commented on tasks/19.html:
```diff
@@ -11,7 +12,50 @@ <h1 class="global-heading">&nbsp;</h1>
+                    background: #ffb366;
+                }
+
+                .global-scope > div:first-child > div
```
у тебя 4 css правила которые красят div в белый, кажется их можно заменить одним

shaikma commented on tasks/20.html:
```diff
@@ -11,7 +12,17 @@ <h1 class="global-heading">&nbsp;</h1>
-                /* your styles here */
+                /* великая магия: https://habrahabr.ru/post/252181/ */
+
+                .red-gt2 div:nth-last-child(n + 3):first-child ~ div
```
зачем `:first-child`?

shaikma commented on tasks/20.html:
```diff
@@ -11,7 +12,17 @@ <h1 class="global-heading">&nbsp;</h1>
+                    background: #ff6698;
+                }
+
+                .red-gt2 div:nth-last-child(n + 3):first-child
```
и здесь? кажется что два правила можно объеденить

shaikma commented on tasks/3.html:
```diff
@@ -11,7 +12,16 @@ <h1 class="global-heading">&nbsp;</h1>
+                div > div
+                {
+                    background: #6698ff;
+                    border: none;
```
попробуй написать стили так, чтобы это строчка не нужна была

shaikma commented on tasks/4.html:
```diff
@@ -11,7 +12,10 @@ <h1 class="global-heading">&nbsp;</h1>
         <main class="global-results">
             <style>
-                /* your styles here */
+                .global-scope article:nth-child(n + 4)
```
тут можно сделать намного проще, посмотри еще раз

shaikma commented on tasks/5.html:
```diff
@@ -11,14 +12,18 @@ <h1 class="global-heading">&nbsp;</h1>
                     <label for="checkbox">Black Theme</label>
                     <input id="checkbox" type="checkbox"><br>
-                    <div class="tab">Text</div>
+                    <div class="tab" id="text">Text</div>
```
на класс сматчиться религиозные убеждения не позволяют?) зачем id?

shaikma commented on tasks/6.html:
```diff
@@ -11,12 +12,20 @@ <h1 class="global-heading">&nbsp;</h1>
         <main class="global-results">
             <style>
-                /* your styles here */
+                article a:first-child
```
используй атрибуты тегов для css

shaikma commented on tasks/7.html:
```diff
@@ -11,7 +12,15 @@ <h1 class="global-heading">&nbsp;</h1>
+                    opacity: .5;
+                }
+
+                img[alt]
```
можно обойтись 1 css правилом

shaikma commented on tasks/9.html:
```diff
@@ -11,7 +12,30 @@ <h1 class="global-heading">&nbsp;</h1>
+                    color: #ff6698;
+                }
+
+                .link[href^='https://ya.ru']:hover + .link
```
1) cсылки независимы друг от друга, у каждой свой внешний вид при наведении, клики и посещения
2) а если url ссылки поменяется? вcе сss правила менять придется

shaikma commented on tasks/8.html:
```diff
@@ -11,7 +12,15 @@ <h1 class="global-heading">&nbsp;</h1>
         <main class="global-results">
             <style>
-                /* your styles here */
+                [rel]
```
лучше указывать тег или класс перед атрибутом

shaikma commented:
🍅

rybufc committed:
исправления

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

shaikma commented on tasks/4.html:
```diff
@@ -12,7 +12,7 @@ <h1 class="global-heading">&nbsp;</h1>
         <main class="global-results">
             <style>
-                .global-scope article:nth-child(n + 4)
+                .global-scope > article:not(:first-of-type)
```
=> `article + article`

shaikma commented on tasks/6.html:
```diff
@@ -12,12 +12,12 @@ <h1 class="global-heading">&nbsp;</h1>
                 }
                 
-                article a:nth-child(n+2)
+                article a[href]
```
`a[href*='yandex']`

shaikma commented on tasks/6.html:
```diff
@@ -12,12 +12,12 @@ <h1 class="global-heading">&nbsp;</h1>
         <main class="global-results">
             <style>
-                article a:first-child
+                article a[href]:first-child
```
а если ссылки поменять местами? лучше так `a[href*='vk']`

shaikma commented on tasks/9.html:
```diff
@@ -12,29 +12,29 @@ <h1 class="global-heading">&nbsp;</h1>
         <main class="global-results">
             <style>
-                .link[href^='https://ya.ru']:hover
+                .link[href]:active
```
попробуй открыть страницу и посмотри срабатывает ли у тебя это правило

shaikma commented on tasks/9.html:
```diff
@@ -12,29 +12,29 @@ <h1 class="global-heading">&nbsp;</h1>
                 }
 
-                .link[href^='https://ya.ru']:active
+                .link:not([href]):active
```
и это тоже

shaikma commented:
🍅

rybufc committed:
4. да, так гораздо проще =)
6. всё понял, тут нужно было не по порядку смотреть элементы, а по содержанию =)
9. исправил

honest-hrundel commented:
:green_apple: Пройден линтинг и базовые тесты

shaikma commented:
@rybufc принял
